### DISCUSSION TITLE
UI theme switch broken?

### DISCUSSION THREAD

**Discussion by Tillerz** [vladmandic/sdnext#1075] [Q&A] [answered]
I can switch, but it just keeps showing the same (default) theme? Where are the themes sitting, anyway? Is it javascipt/ (see below)? I see black-orange in there, but that is doing nothing...

```
.eslintrc.json
aspectRatioOverlay.js
black-orange.css
black-orange.jpg
contextMenus.js
dragdrop.js
edit-attention.js
extensions.js
extraNetworks.js
generationParams.js
gradio-base.jpg
gradio-default.jpg
gradio-glass.jpg
gradio-monochrome.jpg
gradio-soft.jpg
hints.js
hires_fix.js
imageMaskFix.js
imageParams.js
imageviewer.js
notification.js
package.json
progressbar.js
roboto.ttf
script.js
style.css
textualInversion.js
themes.json
ui.js
```

I can see the preview image, but it doesn't apply to the UI at all.

```
10:15:30-795464 INFO     Loading UI theme: black-orange
10:15:30-797965 INFO     Settings changed: 2 ['gradio_theme', 'return_mask']
10:16:14-211478 INFO     Settings changed: 0 []
```

It says it does, but it's not doing a thing.

Right now I am stuck with a bright UI. ![ui](https://i.gyazo.com/a506de1dba25374249a1d35f8237073d.png)

I see there was a post about something similar about the themes, but it seemed to get fixed. I have the latest revision and it's not working for me right now (`Version: 527dc0ee Thu May 18 19:46:11 2023 -0400`).

**Comment by vladmandic** [comment]
only themes that are built-in are default `black-orange`, `gradio/base` and `gradio/default`, the rest are loaded on-demand from huggingface - that's why there is a refresh button next to theme selector so you can fetch available themes list.

i just tried refresh -> switch theme -> apply -> restart and it works.

**Comment by Tillerz** [comment]
Ok, tried it like that with a non-gradio theme (the steampunk one), and that worked.

But then I switched to gradio default and it's back to all bright. Is there a light/dark mode toggle somewhere I do not see or is it following the operating system settings? I can't get it into a dark mode with the gradio/* themes. :-(

PS: also the contrast on the default is really poor, whoever had the idea of darker grey on lighter grey background? That's really not helping with readability.

**Reply by vladmandic** [reply]
those are gradio default themes, nothing to do with me.
does it work in dark mode? no idea. you can try forcing by appending `?__theme=light` or `?__theme=dark` in the url

**Reply by Tillerz** [reply]
Okay, appending that parameter works. Need to find out why light is the default and where it gets it from. :)

**Reply by vladmandic** [reply]
It should get it from browser itself which gets it from OS.

**Reply by vladmandic** [reply]
I can probably add a manual toggle.

**Comment by vladmandic** [comment, marked as answer]
i've added auto/dark/light toggle.

**Reply by Tillerz** [reply]
Thank you so much!
